js/data/bucket.js

```diff
@@ -109,6 +109,35 @@ Bucket.prototype.populateArrays = function() {
 };
 
 /**
+ * Update paint arrays with the given feature properties, leaving geometries
+ * as-is.
+ * @private
+ */
+Bucket.prototype.updatePaintVertexArrays = function(programName, propertiesList) {
```

Comment:
I figured this could be left as the caller's responsibility.

The reason I think this is somewhat reasonable is that `propertiesList` is expected to correspond to the _pre-filtered_ list of features in this tile.  I thought that the ordering of the features in a given vector tile would be stable and safe to rely on.